src/components/accordion/accordion.js

```diff
@@ -0,0 +1,268 @@
+      $headerWrapper.className += ' ' + $module.sectionHeaderFocusedClass
+    }
+  })
+
+  $button.addEventListener('blur', function (e) {
+    $headerWrapper.classList.remove($module.sectionHeaderFocusedClass)
+  })
+
+  if (typeof ($summary) !== 'undefined' && $summary !== null) {
+    $button.setAttribute('aria-describedby', this.moduleId + '-summary-' + (index + 1))
+  }
+
+  // $span could contain HTML elements (see https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content)
+  $button.innerHTML = $span.innerHTML
+
+  $heading.removeChild($span)
+  $heading.appendChild($button)
+
+  // Add "+/-" icon
+  var icon = document.createElement('span')
+  icon.className = this.iconClass
+  icon.setAttribute('aria-hidden', 'true')
+
+  $heading.appendChild(icon)
+}
+
+// When section toggled, set and store state
+Accordion.prototype.onSectionToggle = function ($section) {
+  var expanded = this.isExpanded($section)
+  this.setExpanded(!expanded, $section)
+
+  // Store the state in sessionStorage when a change is triggered
+  this.storeState($section)
+}
+
+// When Open/Close All toggled, set and store state
+Accordion.prototype.onOpenOrCloseAllToggle = function () {
+  var $module = this
+  var $sections = this.$sections
+
+  var nowExpanded = !this.checkIfAllSectionsOpen()
+
+  nodeListForEach($sections, function ($section) {
+    $module.setExpanded(nowExpanded, $section)
+    // Store the state in sessionStorage when a change is triggered
+    $module.storeState($section)
+  })
+
+  $module.updateOpenAllButton(nowExpanded)
+}
+
+// Set section attributes when opened/closed
+Accordion.prototype.setExpanded = function (expanded, $section) {
+  var $button = $section.querySelector('.' + this.sectionButtonClass)
+  $button.setAttribute('aria-expanded', expanded)
+
+  if (expanded) {
+    $section.classList.add(this.sectionExpandedClass)
+  } else {
+    $section.classList.remove(this.sectionExpandedClass)
+  }
+
+  // See if "Open all" button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+// Get state of section
+Accordion.prototype.isExpanded = function ($section) {
+  return $section.classList.contains(this.sectionExpandedClass)
+}
+
+// Check if all sections are open
+Accordion.prototype.checkIfAllSectionsOpen = function () {
+  // Get a count of all the Accordion sections
+  var sectionsCount = this.$sections.length
+  // Get a count of all Accordion sections that are expanded
+  var expandedSectionCount = this.$module.querySelectorAll('.' + this.sectionExpandedClass).length
+  var areAllSectionsOpen = sectionsCount === expandedSectionCount
+
+  return areAllSectionsOpen
+}
+
+// Update "Open all" button
+Accordion.prototype.updateOpenAllButton = function (expanded) {
+  var newButtonText = expanded ? 'Close all' : 'Open all'
+  newButtonText += '<span class="govuk-visually-hidden"> sections</span>'
```

Comment:
Non-blocking suggestion,
These two lines and line 60 look like they could do with a function that returns the mark up.

Also if we wanted to add translation support to the component we could use the macro to determine the text and set a html data attribute which this js could pickup and use.